components/server/src/config/config-inferrer.ts

```diff
@@ -26,11 +26,38 @@ export class ConfigInferrer {
     ];
 
     async getConfig(ctx: Context): Promise<WorkspaceConfig> {
-        for (const contrib of this.contributions) {
-            try {
-                await contrib(ctx);
-            } catch (e) {
-                console.log(e);
+        // initialize cache
+        await Promise.allSettled(
+            [
+                "package.json",
```

Comment:
What is the reasoning behind the additional `allSettled` call here? Wouldn't it be enough to to just go with line 57 here?
:thinking:

Reply:
line 57 only parallelizes the contributions but they are internally still sequentially ask for files. So we would again sequentiallize chunks

Reply:
:ok_hand: 
This way we generate far more requests per API call, though, which in turn might:
 - overloading ourselves
 - consume user's request limits quicker

Let's try it anyway. If one of those pops up, we can still look into investing more here. :+1: